deployments/puppet/README.md

```diff
@@ -0,0 +1,65 @@
+| `splunk_ballast_size_mib` | Set the ballast size for the collector explicitly instead of the value calculated from the `$splunk_memory_total_mib` parameter.  This should be set to 1/3 to 1/2 of configured memory.  The `SPLUNK_BALLAST_SIZE_MIB` environment variable will be set with this value for the collector service. | None |
+| `collector_config_source` | Source path to the collector config YAML file. This file will be copied to the `$collector_config_dest` path on the node.  See the [source attribute](https://puppet.com/docs/puppet/latest/types/file.html#file-attribute-source) of the `file` resource for supported value types.  The default source file is provided by the collector package. | `file:///etc/otel/collector/agent_config.yaml` |
+| `collector_config_dest` | Destination path of the collector config file on the node.  The `SPLUNK_CONFIG` environment variable will be set with this value for the collector service. | `/etc/otel/collector/agent_config.yaml` |
+| `collector_version` | Version of the collector package to install, e.g., `0.25.0`. | `latest` |
+| `service_user` and `$service_group` | **Linux only**: Set the user/group ownership for the collector service. The user/group will be created if they do not exist. | `splunk-otel-collector` |
+| `with_fluentd` | Whether to install/manage fluentd and dependencies for log collection.  The dependencies include [capng_c](https://github.com/fluent-plugins-nursery/capng_c) for enabling [Linux capabilities](https://docs.fluentd.org/deployment/linux-capability), [fluent-plugin-systemd](https://github.com/fluent-plugin-systemd/fluent-plugin-systemd) for systemd journal log collection, and the [required libraries/development tools](https://github.com/signalfx/splunk-otel-collector/blob/main/docs/getting-started/linux-installer.md#fluentd-configuration). | `true` |
+| `fluentd_config_source` | Source path to the fluentd config file.  This file will be copied to the `$fluentd_config_dest` path on the node.  See the [source attribute](https://puppet.com/docs/puppet/latest/types/file.html#file-attribute-source) of the `file` resource for supported value types.  The default source file is provided by the collector package. | `file:///etc/otel/collector/fluentd/fluent.conf` |
+| `fluentd_config_dest` | Destination path to the fluentd config file on the node. | `/etc/otel/collector/fluentd/fluent.conf` |
+| `manage_repo` | **Linux only**: In cases where the collector and fluentd apt/yum repositories are managed externally, set this to `false` to disable management of the repositories by this module.  **Note:** If set to `false`, the apt (`/etc/apt/sources.list.d/splunk-otel-collector.list`, `/etc/apt/sources.list.d/splunk-td-agent.list`) and yum (`/etc/yum.repos.d/splunk-otel-collector.repo`, `/etc/yum.repos.d/splunk-td-agent.repo`) repository definition files will be deleted if they exist in order to avoid any conflicts. | `true` |
```

Comment:
Should we list the expected packages that their repos need to provide if this is false?